ipatests/test_webui/test_service.py

```diff
@@ -352,3 +361,461 @@ def test_kerberos_flags(self):
+        assert self.has_record(pkey)
+
+    @screenshot
+    def test_delete_single_service(self):
+        """
+        Test delete single service
+        """
+        self.init_app()
+        pkey = self.get_service_pkey('single_service')
+        self.add_service('single_service', textbox='service')
+        assert self.has_record(pkey)
+        self.delete_record(pkey)
+        self.assert_notification()
+        assert not self.has_record(pkey)
+
+    @screenshot
+    def test_delete_multiple_services(self):
+        """
+        Test delete multiple services
+        """
+        self.init_app()
+        cifs_pkey = self.get_service_pkey('cifs')
+        nfs_pkey = self.get_service_pkey('nfs')
+        smtp_pkey = self.get_service_pkey('smtp')
+
+        for service in ('cifs', 'nfs', 'smtp'):
+            self.add_service(service)
+
+        for service in (cifs_pkey, nfs_pkey, smtp_pkey):
+            self.delete_record(service)
+            self.wait(0.5)
+            assert not self.has_record(service)
+
+    @screenshot
+    def test_cancel_add_delete_managedby_host(self):
+        """
+        Test cancel/add/delete managed by host
+        """
+        pkey = self.get_service_pkey('HTTP')
+        temp_host = 'host-no-dns.ipa.test'
+        self.init_app()
+
+        # add another host for "managedby" testing
+        self.add_host('host-no-dns', 'ipa.test', force=True)
+
+        self.navigate_to_record(pkey, entity=ENTITY)
+        self.add_associations([temp_host], facet='managedby_host',
+                              confirm_btn='cancel')
+        self.add_associations([temp_host], facet='managedby_host',
+                              delete=True)
+
+        # host cleanup
+        self.navigate_to_entity('host')
+        self.delete_record(temp_host)
+
+    @screenshot
+    def test_add_service_missing_hostname_field(self):
+        """
+        Test add service "hostname" field required
+        """
+        self.init_app()
+        self.navigate_to_entity(ENTITY)
+        self.facet_button_click('add')
+        self.select_combobox('service', 'cifs', combobox_input='service')
+        self.dialog_button_click('add')
+        host_elem = self.find(".widget[name='host']", By.CSS_SELECTOR)
+        self.assert_field_validation_required(parent=host_elem)
+
+    @screenshot
+    def test_add_service_missing_service_field(self):
+        """
+        Test add service "service field required
+        """
+        self.init_app()
+        host = self.config.get('ipa_server')
+        self.navigate_to_entity(ENTITY)
+        self.facet_button_click('add')
+        self.select_combobox('host', host, combobox_input='service')
+        self.dialog_button_click('add')
+        self.wait()
+        service_elem = self.find(".widget[name='service']", By.CSS_SELECTOR)
+        self.assert_field_validation_required(parent=service_elem)
+
+    @screenshot
+    def test_add_service_no_DNS_for_host(self):
```

Comment:
This is the same tests has the same code as test_add_ldap_service - only different service name which is not important.